tasmota/tasmota_xsns_sensor/xsns_02_analog.ino

```diff
@@ -164,6 +167,7 @@ struct {
   uint8_t type = 0;
   uint8_t pin = 0;
   float mq_samples[ANALOG_MQ_SAMPLES];
+  int tds_samples[ANALOG_TDS_SAMPLES];
```

Comment:
This adds up memory which is useless when TDS is not in use
You would save Ram by making a Union of both arrays as they are not used simultaneously

Reply:
I merged mq_samples and tds_samples arrays in one.